### DISCUSSION TITLE
use getServerSideProps in client pages

### DISCUSSION THREAD

**Discussion by askariali** [vercel/next.js#61055] [App Router]
In the **Pages directory,** we could call the `getServerSideProps` function on the server, and its result would be passed down to that page as the props. Is there a way to do that in "use client"-ed pages or, even better, on server components?

**Comment by hyjoong** [comment]
Is it correct that you want to use the SSR method of pages router in the app router as well? If so, you can use fetch(). Fetch allows you to set the caching period, and by adding the `{cache: 'no-store'}` setting, you can implement the SSR of the pages router in the app router as well. fetch('https://...', { cache: 'no-store' }) Using fetch in server components has improved the drawback of having to set the caching period for all communications on one page in the traditional pages router, now allowing you to set the caching period for each server component individually.

Official documentation: https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#individual-fetch-requests
